src/circuit_proof/mod.rs

```diff
@@ -0,0 +1,713 @@
+#![allow(non_snake_case)]
+
+use rand::{CryptoRng, Rng};
+use std::iter;
+
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use curve25519_dalek::traits::{IsIdentity, MultiscalarMul, VartimeMultiscalarMul};
+
+use generators::Generators;
+use proof_transcript::ProofTranscript;  
+use util;
+use inner_product_proof::{inner_product, InnerProductProof};
+
+#[derive(Clone, Debug)]
+pub struct CircuitProof {
+    pub A_I: RistrettoPoint,
+    pub A_O: RistrettoPoint,
+    pub S: RistrettoPoint,
+    pub T_1: RistrettoPoint,
+    pub T_3: RistrettoPoint,
+    pub T_4: RistrettoPoint,
+    pub T_5: RistrettoPoint,
+    pub T_6: RistrettoPoint,
+    pub t_x: Scalar,
+    pub t_x_blinding: Scalar,
+    pub e_blinding: Scalar,
+    pub l_vec: Vec<Scalar>,
+    pub r_vec: Vec<Scalar>,
```

Comment:
Are `l_vec` and `r_vec` supposed to be included in the proof, if we're doing the inner product argument?

Reply:
oh nope - I forgot to remove those after adding the ipp, I'll fix that